BHoM_Engine/Compute/Hash.cs

```diff
@@ -51,7 +51,7 @@ public static partial class Compute
         [Input("typeExceptions", "(Optional) e.g. `typeof(Guid)`. Any corresponding type is ignored.")]
         [Input("maxNesting", "(Optional) e.g. `100`. If any property is nested into the object over that level, it is ignored.")]
         public static string Hash(
-            this IObject iObj,
+            IObject iObj,
```

Comment:
Is there a reason why this input has to be an `IObject` instead of an `object` ?
The `DefiningString` method is already taking in an `object` so might be better to enable it here too.

Reply:
Just for the record (after our discussion yesterday), `DefiningString()` works with `System.Object` because it is made to work recursively and deal with types that may be part (properties) of an `IObject`. It was developed following the structure of the method [`GetValue()`](https://github.com/BHoM/Test_Toolkit/blob/6d230a0ef83bd5a5eebd8b625badb0e2cdd2802d/Test_Engine/Compute/DummyObject.cs#L92-L266) used by `Compute.GetDummy` in the Test_Toolkit (as explained in https://github.com/BHoM/BHoM_Engine/issues/1949). It is therefore guaranteed to work with the limited amount of types included in it (which means, it should work fine with IObjects).  
Extending the parent caller method `Hash()` to object, however, would imply that the Hash could work with any object, which we cannot guarantee.